notebooks/02_model/rbm_deep_dive.ipynb

```diff
@@ -0,0 +1,1449 @@
+    "\n",
+    "$ Z[v,b] = \\prod_j \\sum_{h_j = 0,1}  e^{(\\sum_i w_{ij} v_i + b_j) h_j} = \\prod_j \\left( 1+  e^{\\sum_i w_{ij} v_i + b_j} \\right)$.\n",
+    "\n",
+    "Taking the gradients with respect to the bias we easily obtain \n",
+    "\n",
+    "$\\frac{\\partial}{\\partial b_j}\\log Z[v,b] =  \\frac{1}{1+ e^{-(\\sum_i w_{ij} v_i + b_j)}} = \\sigma( \\phi_j(v, b) )$\n",
+    "\n",
+    "where $\\phi_j(v,b) = \\sum_i w_{ij} v_i + b_j $ and we have identified the logistic function $\\sigma(.) \\equiv P(h_j=1|v,b)$. \n",
+    "\n",
+    "**Use $\\sigma$ to sample the value of $h_j$** \n",
+    "\n",
+    "### Negative \n",
+    "\n",
+    "**Use the sampled value of the hidden units to evaluete P(v_i = q |h)**, where $q=1,...,5$. This is given by the multinomial expression\n",
+    "\n",
+    "$P(v_i = q |h,a) =  \\prod_{v_i=1}^q e^{v_i (\\sum_j w_{ij} \\, h_j + a_i ) }/Z_q $,\n",
+    "\n",
+    "where $Z_q$ is partition function (aka normalization constant) evaluated over the $q$ outcomes (note that $0$ shoul not be included here). Finally, sample the values of $v_i$ from the above distribution. Clearly, these new $v_i$ are not necessarily those we have used as an input, at least not at the beginning of training. The above steps are repeated $k$ times, where $k$ is usually increased during training according to a specified protocol. \n",
+    "\n",
+    "At the end of each k-step Gibbs sampling, we evaluate the difference between the initial free energy ($k=0$) (given v) and the one after k-steps \n",
+    "\n",
+    "$\\Delta F = F_0 - F_k$, \n",
+    "\n",
+    "and update the learning parameters $w_{ij}$, $b_i$ and $a_i$: \n",
+    "\n",
+    "$\\frac{\\partial}{\\partial b_j} \\Delta F = \\frac{\\partial}{\\partial b_j} (\\log Z_0[v,b] - \\log Z_k[v,b]) = P_0(h_j=1|v,b) - P_k(h_j=1|v,b)$\n",
+    "\n",
+    "$\\frac{\\partial}{\\partial w_{ij} } \\Delta F = v_i \\, P_0(v_i = q|h, a) - v_i P_k(v_i| h,a) \\equiv \\langle v_i\\rangle_0 - \\langle v_i \\rangle_k$. \n",
+    "\n",
+    "This process is repeated for each training epoch, eventually until $\\Delta F =0$, i.e. the learned distribution faithfully reproduces the empirical one. In this sense, the $v_i$ serve both as input and output of the model. As $w_{ij}$ contains informations on how users' votes are correlated, we can use this information to generate ratings for the unseen movies by sampling from the learned, marginal distribution:\n",
+    "\n",
+    "$\\langle v_i \\rangle = \\sum_{v_i} v_i \\, P(v)$ \n",
+    "\n",
+    "The entire workflow is summarised below \n",
+    "\n",
+    "![gibbs](https://393229b54263dsvm.file.core.windows.net/images/Gsampling.001.png?sp=rl&st=2018-12-04T08:34:33Z&se=2020-12-05T08:34:00Z&sv=2017-11-09&sig=N0GXdj0UqLsXM2cwIZsWdMgqUTlYvu1pmGhnNmuAOWI%3D&sr=f)\n"
```

Comment:
It is not shown well. Perhaps you can put the image on a blob with public access.

Reply:
I suspect it works on Jupyter because the image file is on the DSVM where Jupyter notebook is hosted. I tried the URL of the image file and it does not show me anything... You can create a new container under an existing storage and authorize open access to the container blob. Ideally the URL of an accessible file is https://<storage_name>.blob.core.windows.net/<container>/<your_file>

Reply:
thanks @yueguoguo. Actually it does visualize well in the jupyter notebook also from other locations other than the dsvm. I often open it on my local machines (laptops, workstations) and still works fine. I suspect it is a problem with git. An easy solution would be  to load the image files on git, as done here 

https://github.com/Microsoft/Recommenders/tree/master/notebooks/04_operationalize

Reply:
+1 to what @yueguoguo said about creating a new container. We have a blob for the repo project called recodatasets, I just created a blob called images. Please let me know if you find it